homeassistant/components/device_tracker/actiontec.py

```diff
@@ -0,0 +1,149 @@
+    for connected devices. Adapted from DD-WRT scanner.
+    """
+
+    def __init__(self, config):
+        self.host = config[CONF_HOST]
+        self.username = config[CONF_USERNAME]
+        self.password = config[CONF_PASSWORD]
+
+        self.lock = threading.Lock()
+
+        self.last_results = {}
+
+        # Test the router is accessible
+        data = self.get_actiontec_data()
+        self.success_init = data is not None
+
+    def scan_devices(self):
+        """ Scans for new devices and return a
+            list containing found device ids. """
+
+        self._update_info()
+        return self.last_results
+
+    def get_device_name(self, device):
+        """ Returns the name of the given device or None if we don't know. """
+        if not self.last_results:
+            return None
+        for client in self.last_results:
+            if client == device:
+                return client
+        return None
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS)
+    def _update_info(self):
+        """ Ensures the information from the Actiontec MI424WR router is up
+            to date. Returns boolean if scanning successful. """
+        if not self.success_init:
+            return False
+
+        with self.lock:
+            # _LOGGER.info("Checking ARP")
+            data = self.get_actiontec_data()
+            if not data:
+                return False
+            active_clients = [client for client in data.values()]
+            self.last_results = active_clients
+            return True
+
+    def get_actiontec_data(self):
+        """ Retrieve data from Actiontec MI424WR and return parsed result.  """
+        try:
+            telnet = telnetlib.Telnet(self.host)
+            telnet.read_until(b'Username: ')
+            telnet.write((self.username + '\n').encode('ascii'))
+            telnet.read_until(b'Password: ')
+            telnet.write((self.password + '\n').encode('ascii'))
+            prompt = telnet.read_until(b'Wireless Broadband Router> ',
+                                       '').split(b'\n')[-1]
+            telnet.write('firewall mac_cache_dump\n'.encode('ascii'))
+            telnet.write('\n'.encode('ascii'))
+            telnet.read_until(prompt)
+            leases_result = telnet.read_until(prompt).split(b'\n')[1:-1]
+            telnet.write('exit\n'.encode('ascii'))
+        except EOFError:
+            _LOGGER.exception("Unexpected response from router")
+            return
+        except ConnectionRefusedError:
+            _LOGGER.exception("Connection refused by router," +
+                              " is telnet enabled?")
+            return
+
+        devices = {}
+        for lease in leases_result:
+            match = _LEASES_REGEX.search(lease.decode('utf-8'))
+            if match is not None:
+                devices[match.group('ip')] = {
+                    'ip': match.group('ip'),
+                    'mac': match.group('mac').upper()

```

Comment:
If this is a dictionary it is not working with your scan_devices method which expects Mac addresses .